### ISSUE TITLE
Context menus not detected via keyboard 

### ISSUE THREAD

**Issue by marius-sucan** [serzh82saratov/AhkSpy#5]
Hello!

In standard win32 windows, controls, for example WinKey+R dialog box, where you have a drop-down and press AppsKey [Menu Key], to invoke the context menu, it is not detected only when you hover on it with the mouse.

Is it possible to detect the menu without hovering it with the mouse?

Thank you .

Best regards, Marius.

**Comment by serzh82saratov** [comment]
У меня работает

**Comment by marius-sucan** [comment]
If I move the mouse cursor, it shows the menu items here as well , but if not, and I just press the menu key, and up / down, nothing seems to show up... Why? 

Best regards, Marius.

**Comment by serzh82saratov** [comment]
Не понимаю что именно не так, сделайте скриншот

**Comment by serzh82saratov** [comment]
Возможно вам лучше задавать вопросы здесь https://www.autohotkey.com/boards/viewtopic.php?f=6&t=52872
Там больше сообщество, и на английском

**Comment by marius-sucan** [comment]
Откройте диалоговое окно. Winkey + R.

Не перемещайте курсор мыши.

Затем нажмите «Меню».

Пункты меню не обнаруживаются.

Вы должны переместить курсор мыши в меню, чтобы они были обнаружены. Почему?

**Comment by serzh82saratov** [comment]
Обнаруживается то что под курсором мыши

**Comment by marius-sucan** [comment]
Есть ли способ обнаружить, что меню открылось? Без наведения мыши. Мне нужно это.

**Comment by serzh82saratov** [comment]
Конечно есть, только причём тут AhkSpy?

**Comment by marius-sucan** [comment]
Я ожидал, что AhkSpy его обнаружит. Подскажите, как это сделать? Spasiba.

**Comment by serzh82saratov** [comment]
что обнаружит? то что какое то окно появилось? зачем это нужно в AhkSpy?

**Comment by marius-sucan** [comment]
Да, просто найди окно. Трудно объяснить почему, по-русски. Spasiba.

**Comment by serzh82saratov** [comment]
Вам надо объяснить мне не почему, а зачем это здесь.
AhkSpy тут не причём, для таких вопросов есть форум.

**Comment by marius-sucan** [comment]
Я ожидал, что AhkSpy сможет это сделать, вот почему.

**Comment by serzh82saratov** [comment]
сделать что, это даже не активное окно.

**Comment by marius-sucan** [comment]
Меню имеет фокус клавиатуры, оно как-то активно ...

**Comment by serzh82saratov** [comment]
не вижу смысла, оно определяется под курсором.
я давно хочу сделать чтобы данные определялись в определённой точке на экране, но не знаю будет ли это реализовано.

**Comment by marius-sucan** [comment]
Я понимаю. Вы можете мне помочь ? Для обнаружения меню без движений мыши.

**Comment by serzh82saratov** [comment]
мне интересно зачем вам это

**Comment by marius-sucan** [comment]
Я пытаюсь улучшить KeyPress OSD. . Мне нужна лучшая поддержка возможностей, связанных с текстом.

**Comment by serzh82saratov** [comment]
как это связано. что именно улучшить KeyPress OSD

**Comment by marius-sucan** [comment]
English:
KeyPress OSD can highlight the location of the text caret on the screen . And I want it to no longer highlight the location if the edit field loses keyboard focus [when a context menu is displayed].

Russian [Google Translate]:
KeyPress OSD может выделить расположение текстовой вставки. И я хочу, чтобы он больше не выделял местоположение курсора, если он теряет фокус клавиатуры [при отображении контекстного меню].

**Comment by serzh82saratov** [comment]
ставьте SetWinEventHook на EVENT_SYSTEM_CAPTURESTART EVENT_SYSTEM_CAPTUREEND смотрите что Accesible Focus имеет
с фокусом ввода и фокусируемое

**Comment by marius-sucan** [comment]
Thank you. I'll try that...

**Comment by serzh82saratov** [comment]
Вы можете использовать это как триггер, но Accesible STATE_SYSTEM_FOCUSED проверять не имеет смысла. Оказалось что не только одно окно может его иметь, для меня это новость. Также не все редактируемые имеют ROLE_SYSTEM_TEXT.

Вам будет проще смотреть что новое окно относится к меню, ждать когда оно будет закрыто.
```AutoHotkey
SetWinEventHook("EVENT_SYSTEM_CAPTURE", EVENT_SYSTEM_FOREGROUND := 3)	
SetWinEventHook("EVENT_SYSTEM_CAPTURE", EVENT_SYSTEM_CAPTURESTART := 8)	
SetWinEventHook("EVENT_SYSTEM_CAPTURE", EVENT_SYSTEM_CAPTUREEND := 9)	 
SetWinEventHook("EVENT_SYSTEM_CAPTURE", EVENT_OBJECT_FOCUS := 0x8005)   
 
EVENT_SYSTEM_CAPTURE(hWinEventHook, event, hwnd, idObject, idChild) { 
	ToolTip EVENT_SYSTEM_CAPTURE
}

SetWinEventHook(EventProc, eventMin, eventMax = 0)  {
	DllCall("CoInitialize", UInt, 0)
	Return DllCall("SetWinEventHook"
				, "UInt", eventMin, "UInt", eventMax := !eventMax ? eventMin : eventMax
				, "Ptr", hmodWinEventProc := 0, "Ptr", lpfnWinEventProc := RegisterCallback(EventProc, "F")
				, "UInt", idProcess := 0, "UInt", idThread := 0
				, "UInt", dwflags := 0x0|0x2, "Ptr")	;;	WINEVENT_OUTOFCONTEXT | WINEVENT_SKIPOWNPROCESS
}
```

**Comment by serzh82saratov** [comment]
Your question is interesting, but you asked from the wrong side at first. :wink:

**Comment by serzh82saratov** [comment]
```AutoHotkey 
DllCall("LoadLibrary", "Str", "oleacc", "Ptr")

SetWinEventHook("EVENT_SYSTEM_CAPTURE", 0x0003, 0x0007)	

EVENT_SYSTEM_CAPTURE(hWinEventHook, event, hwnd, idObject, idChild) {
	If (idObject || idChild)
		Return  
    Static EVENT_SYSTEM_FOREGROUND := 0x0003, EVENT_SYSTEM_MENUSTART := 0x0004
         , EVENT_SYSTEM_MENUEND := 0x0005, EVENT_SYSTEM_MENUPOPUPSTART := 0x0006 
		 , EVENT_SYSTEM_MENUPOPUPEND := 0x0007
		 
	ToolTip % event
}

SetWinEventHook(EventProc, eventMin, eventMax = 0)  {
	; DllCall("CoInitialize", UInt, 0)
	Return DllCall("SetWinEventHook"
				, "UInt", eventMin, "UInt", eventMax := !eventMax ? eventMin : eventMax
				, "Ptr", hmodWinEventProc := 0, "Ptr", lpfnWinEventProc := RegisterCallback(EventProc, "F")
				, "UInt", idProcess := 0, "UInt", idThread := 0
				, "UInt", dwflags := 0x0|0x2, "Ptr")	;;	WINEVENT_OUTOFCONTEXT | WINEVENT_SKIPOWNPROCESS
}
```

**Comment by marius-sucan** [comment]
Spasiba <3. I see you posted two versions. The last one is the final version?

**Comment by serzh82saratov** [comment]
Смотрите сами, это просто подсказки

**Comment by marius-sucan** [comment]
Ох, ну ладно. Большое спасибо
